packages/edit-site/src/components/navigation-inspector/navigation-menu.js

```diff
@@ -56,11 +59,20 @@ export default function NavigationMenu( { innerBlocks, onSelect } ) {
 		} );
 	}, [ updateBlockListSettings, innerBlocks ] );
 
+	// The hidden block is needed because it makes block edit side effects trigger.
+	// For example a navigation page list load its items has an effect on edit to load its items.
```

Comment:
This is unexpected for me and also raises some fundamental questions :) It seems the page list block is using "InnerBlocks" in a very unexpected way 🤔 

 - Is this the only place that does this: fill inner blocks with "edit" functions. I guess not, all "controlled" containers are doing this. And it raises the question of whether this should be part of "edit" function or treated as some kind of "loader" that several blocks can have. 
 - Also if this is some kind of "loader" that is independent of "editing" (edit function), could we leverage this loader to solve the "initial loading of site editor" issue. cc @tyxla @jsnajdr 

Anyway, it doesn't seem like something that needs solving immediately, but it's an interesting question to raise.

Reply:
Yeah, it's an interesting one. I don't think the way you describe it as working is quite right though, instead it generates 'parsed' block data using the pages that exist on a site. I don't think this is a misuse in terms of inner blocks being used as a controlled component, it makes sense in a way that it's synced to an external data source, the same way as reusable blocks or template parts are.

The way list view is evolving though is as a a separate view on a block tree that can now exist without a canvas, and we do have an issue around the block's `edit` encapsulating too much metadata about a block (`allowedBlocks` is already one we know about, but there's also `directInsert` and `renderAppender`). I see the way page list works as similar to those issues.